datadog_checks_dev/datadog_checks/dev/tooling/commands/validate/readmes.py

```diff
@@ -102,13 +102,47 @@ def validate_readme(integration, repo, display_queue, files_failed, readme_count
 
     # Check all required headers are present
     h2s = [h2.text for h2 in soup.find_all("h2")]
-    if "Overview" not in h2s or "Setup" not in h2s:
-        files_failed[readme_path] = True
-        display_queue.append((echo_failure, "     readme is missing either an Overview or Setup H2 (##) section"))
 
-    if "Support" not in h2s and repo == 'marketplace':
+    if "Overview" not in h2s:
+        files_failed[readme_path] = True
+        display_queue.append((echo_failure, "     readme is missing an Overview H2 (##) section"))
+    else:
+        overview_header = soup.find('h2', text='Overview')
+        overview_information = overview_header.find_next()
+        if overview_information.name == "h2":
+            files_failed[readme_path] = True
+            display_queue.append((echo_failure, "     readme has an empty Overview H2 (##) section"))
+
+    if "Setup" not in h2s:
         files_failed[readme_path] = True
-        display_queue.append((echo_failure, "     readme is missing a Support H2 (##) section"))
+        display_queue.append((echo_failure, "     readme is missing a Setup H2 (##) section"))
+    else:
+        setup_header = soup.find('h2', text='Setup')
+        setup_instructions = setup_header.find_next()
+        if setup_instructions.name == "h2":
+            files_failed[readme_path] = True
+            display_queue.append((echo_failure, "     readme has an empty Setup H2 (##) section"))
```

Comment:
I think you could do something like this:

```
for section in ("Overview", "Setup"):
  if section not in h2s:
          files_failed[readme_path] = True
          display_queue.append((echo_failure, f"     readme is missing a {section} H2 (##) section"))
      else:
          setup_header = soup.find('h2', text=section)
          setup_instructions = setup_header.find_next()
          if setup_instructions.name == "h2":
              files_failed[readme_path] = True
              display_queue.append((echo_failure, f"     readme has an empty {section} H2 (##) section"))
```

Reply:
In fact you could even extract the logic in the for loop to its own function

Reply:
Thanks fixed it! looks cleaner